jetson/nav/search/searchStateMachine.cpp

```diff
@@ -213,19 +236,40 @@ NavState SearchStateMachine::executeTurnToTarget()
 {
     if( mRover->roverStatus().leftTarget().distance < 0 )
     {
-        cerr << "Lost the target. Continuing to turn to last known angle\n";
-        if( mRover->turn( mTargetAngle + mTurnToTargetRoverAngle ) )
-        {
-            return NavState::TurnedToTargetWait;
+        // TODO: Clean this up and/or move it to avoid duplication
+        if ( mRover->roverStatus().leftCacheTarget().distance != -1 ) {
+            if( mRover->turn( mRover->roverStatus().leftCacheTarget().bearing +
+                      mRover->roverStatus().odometry().bearing_deg ) )
+            {
+                return NavState::DriveToTarget;
+            }
+            updateTargetDetectionElements( mRover->roverStatus().leftCacheTarget().bearing,
+                                       mRover->roverStatus().odometry().bearing_deg );
+            return NavState::TurnToTarget;
+        }
+        else {
+            cerr << "Lost the target. Continuing to turn to last known angle\n";
+            if( mRover->turn( mTargetAngle + mTurnToTargetRoverAngle ) )
```

Comment:
is the last known angle being updated when using a cached target?

Reply:
Yes, in the above lines for when we can't find the current target (function `updateTargetDetectionElements` on line 246 updates the turn so that in the event we lose it, it will turn to the cached one instead)

Reply:
great, thx